docs/help.md

```diff
@@ -0,0 +1,241 @@
+# Help!
+
+**This document lists out all errors and warnings in `@urql/exchange-graphcache`.**
+
+Any unexpected behaviour, conditon, or error will be marked by an error or warning
+in development, which will output a helpful little message. Sometimes however, this
+message may not actually tell you everything about what's going on.
+
+This is a supporting document that explains every error and attempts to give more
+information on how you may be able to fix some issues or avoid these errors/warnings.
+
+## (1) Invalid GraphQL document <a id="1"></a>
+
+> Invalid GraphQL document: All GraphQL documents must contain an OperationDefinition
+> node for a query, subscription, or mutation.
+
+There are multiple places where you're passing in GraphQL documents, either through
+methods on `Cache` (e.g. `cache.updateQuery`) or via `urql` using the `Client` or
+hooks like `useQuery`.
+
+Your queries must always contain a main operation, so either a query, mutation, or
+subscription. This error occurs when this is missing, because the `DocumentNode`
+is maybe empty or only contains fragments.
+
+## (2) Invalid Cache call <a id="2"></a>
+
+> Invalid Cache call: The cache may only be accessed or mutated during
+> operations like write or query, or as part of its resolvers, updaters,
+> or optimistic configs.
+
+If you're somehow accessing the `Cache` (an instance of `Store`) outside of any
+of the usual operations then this error will be thrown. Typically you'd only
+use the cache inside resolvers that you pass to the `cacheExchange` as a
```

Comment:
just for clarity I wouldn't assume that our user accounts updates, optimistic, ... as resolvers since resolvers is an explicit configuration